iree/compiler/Conversion/VectorToLLVM/test/vector_contract_to_aarch64_asm.mlir

```diff
@@ -0,0 +1,42 @@
+// RUN: iree-opt -iree-codegen-vector-to-aarch64-inline-asm %s | IreeFileCheck %s
+
+func @vector_matmul_to_aarch64_asm_vec_dot(%lhs: memref<4x4xi8>, %rhs: memref<4x4xi8>, %dst: memref<4x4xi32>) {
+    %c0 = constant 0 : index
+    %cst_i8_0 = constant 0 : i8
+    %cst_i32_0 = constant 0 : i32
+    %0 = vector.transfer_read %lhs[%c0, %c0] , %cst_i8_0 {masked = [false, false]} : memref<4x4xi8>, vector<4x4xi8>
+    %1 = vector.transfer_read %rhs[%c0, %c0] , %cst_i8_0 {masked = [false, false]} : memref<4x4xi8>, vector<4x4xi8>
+    %2 = vector.transfer_read %dst[%c0, %c0], %cst_i32_0 {masked = [false, false]} : memref<4x4xi32>, vector<4x4xi32>
+    %3 = sexti %0 : vector<4x4xi8> to vector<4x4xi32>
+    %4 = sexti %1 : vector<4x4xi8> to vector<4x4xi32>
+    %5 = vector.contract {
+        indexing_maps = [
+            affine_map<(d0, d1, d2) -> (d0, d2)>,
+            affine_map<(d0, d1, d2) -> (d2, d1)>,
+            affine_map<(d0, d1, d2) -> (d0, d1)>
+        ], iterator_types = ["parallel", "parallel", "reduction"], kind = #vector.kind<add>
+    } %3, %4, %2 : vector<4x4xi32>, vector<4x4xi32> into vector<4x4xi32>
+    vector.transfer_write %5, %dst[%c0, %c0] {masked = [false, false]}: vector<4x4xi32>, memref<4x4xi32>
+    return
+}
+// CHECK-LABEL: vector_matmul_to_aarch64_asm_vec_dot
+// CHECK: %[[RES_2D:.+]] = constant dense<0> : vector<4x4xi32>
```

Comment:
Lets try some consistent FileCheck writing in IREE (at least for codegen parts). I'd prefer to 

1) Align all the `:`.
2) Use Check-DAG wherever a sequence of operations dont have a specific order.